### ISSUE TITLE
Different focus ring behavior in Firefox and Chrome

### ISSUE THREAD

**Issue by darekkay** [palantir/blueprint#3146]
#### Environment

- __Package version(s)__: 3.8.0
- __Browser and OS versions__: Firefox 63, Chrome 70

#### Steps to reproduce

Press TAB key on Buttons and AnchorButtons on the [demo page](https://blueprintjs.com/docs/#core/components/button) to display the focus ring (focus indicator)

#### Actual behavior

Default browser focus ring is visible in Firefox for normal Buttons and a dedicated focus ring is displayed for AnchorButtons. In Chrome, a dedicated focus ring is visible for both Buttons.

![image](https://user-images.githubusercontent.com/3101914/48373771-16c7bc80-e6c3-11e8-976e-15c8289599b1.png)

#### Expected behavior

Same focus ring should be displayed for Buttons and AnchorButtons in Firefox.

**Comment by giladgray** [comment]
@darekkay this is expected, as browser support for CSS outline is different.

**Comment by darekkay** [comment]
@giladgray This is not about the browser differences. It's about inconsistent focus indicitator in Firefox. It is displayed correctly for an AnchorButton and not correctly for a Button, hence this is an implementation issue and not linked to the browser support.

**Comment by giladgray** [comment]
i see that dashed line on both button types in firefox, but if you set `focus` in the inspector then the blue ring appears. seems firefox does its own focus rendering, so it is a browser thing.
